src/main/java/com/powsybl/openloadflow/network/util/ActivePowerDistribution.java

```diff
@@ -104,4 +108,60 @@ public static Step getStep(LoadFlowParameters.BalanceType balanceType, boolean l
                     new GenerationActivePowerDistributionStep(GenerationActivePowerDistributionStep.ParticipationType.REMAINING_MARGIN, useActiveLimits);
         };
     }
+
+    public record ResultAfterFailureBehaviorHandling(boolean failed, String failedMessage, int iteration, double remainingMismatch, boolean movedBuses, double additionalDistributedActivePower) { }
+
+    public static ResultAfterFailureBehaviorHandling handleDistributionFailureBehavior(OpenLoadFlowParameters.SlackDistributionFailureBehavior behavior,
+                                                                                       LfGenerator referenceGenerator,
+                                                                                       double requestedActivePower,
```

Comment:
Something like `slackActivePowerInitialMismatch` is more explicit than `requestedActivePower`?

Reply:
changed to `activePowerMismatch` just like in the other `run(...)` method